scripts/sanity_chk/sanitylib.py

```diff
@@ -2783,8 +2806,14 @@ def apply_filters(self, **kwargs):
                     continue
 
                 if plat.ram < tc.min_ram:
-                    discards[instance] = "Not enough RAM"
-                    continue
+                    reason = "Not enough RAM"
+                    discards[instance] = reason
+                    instance.reason = reason
+                    instance.status = "skipped"
+                    self.total_skipped += 1
+                    self.total_skipped_cases += len(tc.cases)
+                    instance.fill_results_by_status()
+                    # continue
```

Comment:
why dont you work on the resulting discards dictionary instead of doing this one by one here for each filter?

Reply:
Haven't thought about it :sweat_smile:

Reply:
How should we treat the msg for skips? Does the order in which skips/continues were happening matter? If so, the msg can be filled with only the first occurring reason. Or is it better to list all the reasons?

Reply:
the first reason wins, the order is not significant.